onnxruntime/test/python/onnxruntime_test_python.py

```diff
@@ -1065,20 +1065,30 @@ def check_failure(providers, provider_options):
         check_failure([("a", {1: 2})], [{3: 4}])
 
     def testRegisterCustomEPsLibrary(self):
-        # exclude for macos and linux
-        if not sys.platform.startswith("win"):
+        from onnxruntime.capi import _pybind_state as C
+        available_eps = C.get_available_providers()
+        #skip amd gpu build
+        if 'kRocmExecutionProvider' in available_eps:
+            return
+        if sys.platform.startswith("win"):
+            shared_library = 'test_execution_provider.dll'
+            if not os.path.exists(shared_library):
+                raise FileNotFoundError("Unable to find '{0}'".format(shared_library))
+
+        elif sys.platform.startswith("darwin"):
+            # exclude for macos
             return
 
-        shared_library = 'test_execution_provider.dll'
-        if not os.path.exists(shared_library):
-            raise FileNotFoundError("Unable to find '{0}'".format(shared_library))
-        
+        else:
+            shared_library = './libtest_execution_provider.so'
+            if not os.path.exists(shared_library):
```

Comment:
Can this be shared with the windows case? The error checks look identical for both.

Reply:
yes